flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamTaskStateInitializerImpl.java

```diff
@@ -266,6 +266,11 @@ protected OperatorStateBackend operatorStateBackend(
 			taskInfo.getNumberOfParallelSubtasks(),
 			taskInfo.getIndexOfThisSubtask());
 
+		// Now restore processing is included in backend building/constructing process, so we need to make sure
+		// each stream constructed in restore could also be closed in case of task cancel, for example the data
+		// input stream opened for serDe during restore.
+		CloseableRegistry cancelStreamRegistryForRestore = new CloseableRegistry();
+		backendCloseableRegistry.registerCloseable(cancelStreamRegistryForRestore);
```

Comment:
It seems like there is still no matching `unregister` call for this, so that means in theory `cancelStreamRegistryForRestore` is never removed from `backendCloseableRegistry` until ``backendCloseableRegistry` is closed, right? But I would expect that we can close and remove `cancelStreamRegistryForRestore` after calling `createAndRestore` now.